front/app/components/admin/GraphCards/_utils/query.ts

```diff
@@ -77,3 +79,30 @@ export const getDateFilterLastPeriod = (
     },
   };
 };
+
+export const getComparedTimeRange = (startAt: string, endAt: string) => {
+  if (!startAt || !endAt) return {};
+
+  const startAtMoment = moment(startAt, 'YYYY-MM-DD');
+  const endAtMoment = moment(endAt, 'YYYY-MM-DD');
+
+  const days = endAtMoment.diff(startAtMoment, 'days');
+
+  const prevEndAtMoment = startAtMoment.clone().subtract(1, 'days');
+  const prevStartAtMoment = prevEndAtMoment.clone().subtract(days, 'days');
```

Comment:
Maybe in this case not, but generally when you use methods like `subtract` it's good practice to `clone` first, because `moment` has the horribly annoying feature of in some cases mutating the old object in place in addition to returning whatever the method should return